bubble_sort_by.rb

```diff
@@ -0,0 +1,20 @@
+# creating bubble sort that accept a block
+
+def bubble_sort_by(array)
+  n = array.length - 1
+  (array.length - 1).times do
+    switched = false
+    n.times do |i|
+      if yield(array[i], array[i + 1]).positive?
+        array[i], array[i + 1] = array[i + 1], array[i]
+        switched = true
+      end
+    end
+    break unless switched
+
+    n -= 1
+  end
+  [array, array.length - n]
+end
```

Comment:
- [ ] bubble_sort_by method should be working **(covered by automated tests)** ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
- [ ] bubble_sort_by sorts an array by accepting a block
```ruby
words = %w[hi hey hello]
bubble_sort_by(words) { |left, right| left.length - right.length }
```
Should return `["hi", "hey", "hello"]` but instead it returns `[["hi", "hey", "hello"], 2]`